recipes/natural_language_processing/chatbot-java-quarkus-rag/app/Containerfile

```diff
@@ -0,0 +1,12 @@
+FROM registry.access.redhat.com/ubi8/openjdk-21:latest
+WORKDIR /deployments
+COPY --chown=185 . .
+RUN mvn package
+EXPOSE 8080
+ENV JAVA_OPTS="-Dquarkus.http.host=0.0.0.0 -Djava.util.logging.manager=org.jboss.logmanager.LogManager"
+ENV JAVA_APP_JAR="/deployments/target/quarkus-app/quarkus-run.jar"
+# FIXME: the sleep is to allow the postgresql container to start first and
+# should be replaced with a mechanism to declare that the application
+# container depends on the database container, once something like that is
+# supported...
+CMD echo "sleeping for 10 seconds..."; sleep 10; /usr/local/s2i/run
```

Comment:
The recipes should following the cloud native pattern so should implement retry if the targeted service is not available or handle propertly failures so is this required ?

Reply:
Unfortunately, Quarkus itself does not support retries like this (it relies on it being restarted if the database isn't ready yet) and just refuses to boot if it can't connect :( I know this is a terrible solution, but I don't have any better idea.

Reply:
Or is it possible to declare somewhere (in `ai-lab.yaml`) that the Quarkus pod should be restarted until it boots properly?

Reply:
No because the design has been made so that there is no synchronisation between the containers that are started. There is no way to delay the connection to the database ?

Reply:
@yrodiere could you advise if there is a way to let Quarkus boot even when the database is not available yet, and let Hibernate execute its database generation strategy more lazily? I checked the configuration properties and nothing came to my mind.

Reply:
Ok, in any case, none of this looks like something we want to do in this recipe that should be kept as simple as possible. IMO, AI lab should provide a way to declare dependencies between pods. Without it,  I have no better idea than a dumb sleep command

Reply:
Hmm, or maybe, add some piece of a shell script to wait for the database port to be open before calling `/usr/local/s2i/run`??